src/main/java/jsr223/perl/PerlScriptEngineFactory.java

```diff
@@ -0,0 +1,132 @@
+ * or a different license than the AGPL.
+ */
+package jsr223.perl;
+
+import java.util.*;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineFactory;
+
+import jsr223.perl.utils.PerlVersionGetter;
+import processbuilder.PerlSingletonPerlProcessBuilderFactory;
+import processbuilder.utils.PerlProcessBuilderUtilities;
+
+
+public class PerlScriptEngineFactory implements ScriptEngineFactory {
+
+    // Script engine parameters
+    private static final String NAME = "perl";
+
+    private static final String ENGINE = NAME;
+
+    private static final String ENGINE_VERSION = "0.1.0";
+
+    private static final String LANGUAGE = "perl";
+
+    private final Map<String, Object> parameters = new HashMap<>();
+
+    private static PerlProcessBuilderUtilities processBuilderUtilities = new PerlProcessBuilderUtilities();
+
+    private static PerlVersionGetter perlVersionGetter = new PerlVersionGetter(processBuilderUtilities);
+
+    public PerlScriptEngineFactory() {
+        parameters.put(ScriptEngine.NAME, NAME);
+        parameters.put(ScriptEngine.ENGINE_VERSION, ENGINE_VERSION);
+        parameters.put(ScriptEngine.LANGUAGE, LANGUAGE);
+        parameters.put(ScriptEngine.ENGINE, ENGINE);
+    }
+
+    public PerlScriptEngineFactory(PerlProcessBuilderUtilities processBuilderUtilities,
+            PerlVersionGetter perlVersionGetter) {
+        this();
+        if (processBuilderUtilities == null || perlVersionGetter == null) {
+            throw new NullPointerException("processBuilderUtilities and perlVersionGetter must not be null");
+        }
+        this.processBuilderUtilities = processBuilderUtilities;
+        this.perlVersionGetter = perlVersionGetter;
+
+    }
+
+    @Override
+    public String getEngineName() {
+        return (String) parameters.get(ScriptEngine.NAME);
+    }
+
+    @Override
+    public String getEngineVersion() {
+        return (String) parameters.get(ScriptEngine.ENGINE_VERSION);
+    }
+
+    @Override
+    public List<String> getExtensions() {
+        return Arrays.asList("pl", "pm", "t", "pod");
+    }
+
+    @Override
+    public List<String> getMimeTypes() {
+        return Collections.singletonList("application/x-perl");
+    }
+
+    @Override
+    public List<String> getNames() {
+        return Arrays.asList(ENGINE, "pearl", "Perl");
+    }
+
+    @Override
+    public String getLanguageName() {
+        return (String) parameters.get(ScriptEngine.LANGUAGE);
+    }
+
+    @Override
+    public String getLanguageVersion() {
+        return perlVersionGetter.getPerlVersion(PerlSingletonPerlProcessBuilderFactory.getInstance());
+    }
+
+    @Override
+    public Object getParameter(String key) {
+        return parameters.get(key);
+    }
+
+    @Override
+    public String getMethodCallSyntax(String obj, String m, String... args) {
+        return null;
+    }
+
+    @Override
+    public String getOutputStatement(String toDisplay) {
```

Comment:
you can have a look at http://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngineFactory.html to learn how to implement these methods